src/__tests__/server.test.ts

```diff
@@ -72,37 +100,10 @@ describe("Server graceful shutdown", () => {
     app.emit("server:stop");
   });
 
-  it("should wait requests on test route to complete before shitting down the server", async () => {
-    const neverCalledFunction = jest.fn();
-    const completeAsyncOp = jest.fn();
-
-    // Call test api after SIGTERM and expect that the response will be refused
-    setTimeout(async () => {
-      // Start Server shutdown
-      process.emit("SIGTERM", "SIGTERM");
-      try {
-        await nodeFetch(testApiUrl);
-        neverCalledFunction(); // test API will never succeded
-      } catch (e) {
-        expect(e.code).toBe("ECONNREFUSED");
-        completeAsyncOp(); // check that the test API return an error
-      }
-    }, 10);
-
-    const response = await nodeFetch(testApiUrl);
-    expect(await response.json()).toEqual(expectedResponse);
-
-    // Check that the shutting down process has been completed after the timeout value.
-    return await new Promise<void>(resolve => {
-      setTimeout(() => {
-        // Finally is called twice, one when the stack of connections become zero and one when the graceful shutdown timeout is reached.
-        expect(finallyMock).toBeCalledTimes(2);
-
-        // check that the second test API call never succeded
-        expect(neverCalledFunction).not.toBeCalled();
-        expect(completeAsyncOp).toBeCalledTimes(1);
-        resolve();
-      }, gracefulShutdownTimeout);
-    });
+  it("should call finally functions in HttpGracefulShutdown two times", async () => {
+    process.emit("SIGTERM", "SIGTERM");
+    setTimeout(() => {
+      expect(finallyMock).toHaveBeenCalled();
+    }, gracefulShutdownTimeout);
```

Comment:
If we have to refactor this test as a Unit Test, avoiding async calls ( see previous comment ), do we have to refactor the entire test, removing app instantiation and simply focus on `initGracefulShutdown(...)` function, by providing a test that check if finally function is called ?